### PR TITLE
lifetimes: try to fix w.r.t. lifetimes from parent scopes (fixes #162)

### LINKED ISSUE #162: Elision lint should bail if we're using named lifetimes defined *outside* the function
This is for methods which use named lifetimes which are defined elsewhere (not in `fn foo<'a, 'b>(..`), e.g. https://github.com/Manishearth/rust-clippy/blob/master/src/loops.rs#L106

I think when we encounter such lifetimes we can only bail, because the lifetime may or may not be connected with the self type and elision can't do much anyway.

cc @birkenfeld

### PR DESCRIPTION
**Pull request by birkenfeld**

### CODE REVIEW

**Comment by birkenfeld** [comment]
Q: should we check MethodTraitItems?

File: src/lifetimes.rs [hunk 1]
```diff
@@ -39,10 +48,34 @@ enum RefLt {
...
     // There are two scenarios where elision works:
     // * no output references, all input references have different LT
     // * output references, exactly one input reference with same LT
+    // All lifetimes must be unnamed, 'static or defined without bounds on the
+    // level of the current item.
+
+    // check named LTs
+    let mut allowed_lts = HashSet::new();
+    for lt in named_lts {
+        if lt.bounds.is_empty() {
```
Comment (Manishearth) [inline src/lifetimes.rs]: if bounds is not empty, return

**Comment by Manishearth** [comment]
r+, but subsumed into #160

**Comment by Manishearth** [comment]
MethodTraitItems are non-self methods, so yes. Preferably in a followup PR, I subsumed this into the dogfood one

**Comment by birkenfeld** [comment]
ok, closing this one.

**Comment by Manishearth** [comment]
I was planning on leaving it open, it would be marked as automerged when dogfood merged, but whatever :)

**Pull request closed without merging**